### DISCUSSION TITLE
How can I crate a cache without any eviction policies

### DISCUSSION THREAD

**Discussion by alizhan1** [ben-manes/caffeine#874] [Q&A] [answered]
Hi, guys!
I was implementing cache in one of my services, and it requires that all entries will stay in cache, unless the delete method is called explicitly in the code, meaning that built in eviction policies of Caffeine should never be triggered.

```
@Bean
public Caffeine<Object, Object> caffeineForA()
{
    return Caffeine.newBuilder();
}
```

```
@Bean
public CacheManager cacheManagerForA(final Caffeine<Object, Object> caffeineForA)
{
    final CaffeineCacheManager caffeineCacheManager = new CaffeineCacheManager();
    caffeineCacheManager.setCaffeine(caffeineForA);
    return caffeineCacheManager;
}
```

I expect this configuration to work like this. If it is not correct, can you suggest to me how can I achieve the things I mentioned earlier.
Thanks

**Comment by ben-manes** [comment, marked as answer]
Yes, that should be fine. If you don’t specify a policy then we use a light adapter from our interfaces to ConcurrentHashMap. Spring similarly offers ConcurrentMapCache if you prefer.
